mongo-object.js

```diff
@@ -275,8 +275,7 @@ MongoObject = function(objOrModifier, blackBoxKeys) {
         if (current[subkey] === void 0 && value !== void 0) {
           //see if the next piece is a number
           nextPiece = subkeys[i + 1];
-          nextPiece = parseInt(nextPiece, 10);
-          current[subkey] = isNaN(nextPiece) ? {} : [];
+          current[subkey] = isNaN(+nextPiece) ? {} : [];
```

Comment:
I did not pay attention to the `expandKey` method because it has no usages not in `SimpleSchema`, not in my project.